s2tbx-cache/src/main/java/org/esa/s2tbx/dataio/cache/S2CacheSizeChecking.java

```diff
@@ -0,0 +1,69 @@
+package org.esa.s2tbx.dataio.cache;
+
+import java.io.File;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.esa.snap.core.util.ThreadExecutor;
+import org.esa.snap.core.util.ThreadRunnable;
+
+public class S2CacheSizeChecking {
+
+    public static final S2CacheSizeChecking INSTANCE = new S2CacheSizeChecking();
+    private static final Logger logger = Logger.getLogger(S2CacheSizeChecking.class.getName());
+    private boolean checkingEnable;
+    private double limitSizeCache;
+    private S2CacheSizeChecking(){
+        //nothing to init
+    }
+    
+    public static S2CacheSizeChecking getInstance() {
+        return INSTANCE;
+    }
+    /**
+     * set the parameters of the cache size checking
+     *
+     */
+    public synchronized void setParameters(boolean checkingEnable,double limitSizeCache ) {
+        this.checkingEnable = checkingEnable;
+        this.limitSizeCache = limitSizeCache;
+    }
+
+    /**
+     * Method check the S2cache size over time. The method is executed in thread.
+     * Each period time, the cache size is checked.
+     * The x% of the oldest files should be deleted in case of cache oversize.
+     * @param releasedSpacePurcent The space purcentage of the oldest files should be deleted
+     * @param period The period between two size checking in minute
+     */
+    public synchronized void launchCacheSizeChecking(double releasedSpacePurcent, int period) {
+        final ThreadExecutor executor = new ThreadExecutor();
+            try {
+                ThreadRunnable runnable = new ThreadRunnable() {
+                    @Override
+                    public void process() throws Exception {
+                        double byteToGigaByte = 0.000000001;
+                        while(!Thread.currentThread().isInterrupted()){
```

Comment:
This is practically an infinite loop, `Thread.isInterrupted()` probably isn't that useful here. The code could use a condition variable to wait for exit notification, but setting up a timer would be better (and wouldn't hold up a thread pool thread indefinitely).

In addition, it looks like this _could_ presumably remove files which aren't in use, which might not be great.

Reply:
Your suggestions might work, but actually this is started from S2CacheActivator.start() and should be stopped by S2CacheActivator.stop()

Reply:
I suppose the `S2CacheActivator` might just as well cancel the timer.